### PR TITLE
Feature/ Unit tests for Constructor, Driver and Login

### PR DESCRIPTION
**Pull request by ksookdhew**
## Description 
Created the unit tests for Constructor, Driver and Login

## Checklist 
- [x] The code follows the project's coding standards. 
- [x] 100% Unit test coverage for View Models 
- [x] All warnings Resolved.

### CODE REVIEW

**Review by giacatano — DISMISSED** [review]

**Review by AShozi — DISMISSED** [review]

File: F1/F1/Login/LoginViewModel.swift [hunk 1]
```diff
@@ -8,16 +8,17 @@
...
+
+    // MARK: Variables
     private let username = "Admin"
     private let password = "Password"
 
-    func validDetails(givenUsername: String, givenPassword: String) -> Bool{
+    // MARK: Functions
+    func validDetails(givenUsername: String, givenPassword: String) -> Bool {
         if givenUsername==username && givenPassword==password {
             return true
```
Comment (kieran-woodrow) [inline F1/F1/Login/LoginViewModel.swift:19]: try avoid these if something return true else return false
You can make it so much cleaner and just say 
return givenUsername == username && givenPassword == password

File: F1/F1/Constructor/ConstructorViewModel.swift [hunk 2]
```diff
@@ -18,22 +19,26 @@ class ConstructorViewModel {
...
     }
 
     var drivers: String {
-        return "\(constructorResults?.first?.results.first?.driver.code ?? "") | \(constructorResults?.first?.results.last?.driver.code ?? "")"
+        "\(constructorResults?.first?.results.first?.driver.code ?? "") | \(constructorResults?.first?.results.last?.driver.code ?? "")"
     }
 
+    // MARK: Functions
+    func result(atIndex: Int) -> Race? {
+        if constructorResults?.count ?? 0 > atIndex, let result = constructorResults {
```
Comment (kieran-woodrow) [inline F1/F1/Constructor/ConstructorViewModel.swift:34]: func result(atIndex: Int) -> Race? {
   constructorResults?.count ?? 0 > atIndex ? constructorResults?[atIndex] : nil
}

removes a lot of lines

File: F1/F1/Driver/DriverViewModel.swift [hunk 3]
```diff
@@ -19,10 +20,12 @@ class DriverViewModel {
         self.delegate = delegate
     }
 
+    // MARK: Computed Variables
```
Comment (kieran-woodrow) [inline F1/F1/Driver/DriverViewModel.swift:23]: can remove return in computed property

File: F1/F1/Driver/DriverViewModel.swift [hunk 4]
```diff
@@ -19,10 +20,12 @@ class DriverViewModel {
...
     var resultsCount: Int {
         return driverResults?.count ?? 0
     }
 
+    // MARK: Functions
```
Comment (kieran-woodrow) [inline F1/F1/Driver/DriverViewModel.swift:28]: same in this function

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: F1/F1/Constructor/ConstructorViewModel.swift [hunk 5]
```diff
@@ -18,22 +19,23 @@ class ConstructorViewModel {
...
-        return nil
+    var drivers: String {
+        "\(constructorResults?.first?.results.first?.driver.code ?? "") | \(constructorResults?.first?.results.last?.driver.code ?? "")"
     }
 
-    var drivers: String {
-        return "\(constructorResults?.first?.results.first?.driver.code ?? "") | \(constructorResults?.first?.results.last?.driver.code ?? "")"
+    // MARK: Functions
+    func result(atIndex: Int) -> Race? {
+    constructorResults?.count ?? 0 > atIndex ? constructorResults?[atIndex] : nil
```
Comment (kieran-woodrow) [inline F1/F1/Constructor/ConstructorViewModel.swift:34]: indenting..

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: F1/F1/Constructor/ConstructorViewModel.swift (hunk 5 above)
Comment (ksookdhew) [reply F1/F1/Constructor/ConstructorViewModel.swift:34]: 🤣

**Review by kieran-woodrow — APPROVED** [review]

**Pull request merged**